FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift

```diff
@@ -43,33 +43,54 @@ class ModelDownloadTask: NSObject {
   private let appName: String
   /// Model info downloaded from server.
   private(set) var remoteModelInfo: RemoteModelInfo
+  /// Model conditions for download.
+  private let conditions: ModelDownloadConditions
   /// User defaults to which local model info should ultimately be written.
   private let defaults: UserDefaults
   /// Task to handle model file download.
   private var downloadTask: URLSessionDownloadTask?
   /// Progress and completion handlers associated with this model download task.
   private let downloadHandlers: DownloadHandlers
   /// Keeps track of download associated with this model download task.
-  private(set) var downloadStatus: ModelDownloadStatus = .notStarted
+  private(set) var downloadStatus: ModelDownloadStatus
+  /// Background session identifier.
+  private let backgroundSessionIdentifier = "ModelDownloadBackgroundSession"
+  /// Optional background completion handler.
+  private var backgroundCompletion: (() -> Void)?
   /// URLSession to handle model downloads.
-  private lazy var downloadSession = URLSession(configuration: .ephemeral,
-                                                delegate: self,
-                                                delegateQueue: nil)
+  private lazy var downloadSession: URLSession = {
+    var configuration = URLSessionConfiguration.ephemeral
+    if conditions.allowsBackgroundDownloading {
+      configuration = URLSessionConfiguration
```

Comment:
Is this less secure than `.ephemeral`?

Reply:
Yes, it may save unencrypted request/response in cache database in the app sandbox. It is not necessarily a bit issue but is safer to avoid if possible.